### ISSUE TITLE
Add Support for MQTT

### ISSUE THREAD

**Issue by ashitaprasad** [foss42/apidash#115]
**Tell us about the task you want to perform and are unable to do so because the feature is not available**
See related discussion point 2 in https://github.com/foss42/apidash/discussions/112

**Comment by CodeHariK** [comment]
do you have figma design how you will integrate mqtt ui

**Comment by animator** [comment]
No @CodeHariK 
This involves end to end feature build right from the UI design stage to architecture and implementation.

**Comment by mmjsmohit** [comment]
I would like to work on this issue. If anyone is already working on this issue, please let me know.

**Comment by mmjsmohit** [comment]
@animator @ashitaprasad Since this issue comes under the purview of realtime protocols, should I wait for #15 to close or submit a PR for this issue as well?

**Comment by animator** [comment]
You can submit a PR @mmjsmohit

**Comment by mmjsmohit** [comment]
@animator @ashitaprasad Here is the current progress on this issue. Would love to receive suggestions on the same.

[Progress.webm](https://github.com/foss42/apidash/assets/65406090/e2da151a-1b57-4c80-a4cd-b4124c43dd59)

For now, I have hardcoded the topic as `helloWorld` for testing purposes. Will add a `TextField` to input the topic name besides the send button for that.

**Comment by badnikhil** [comment]
@animator the project in my 2nd proposal includes adding mqtt and grpc support. Is it fine if I implement that now? Or I should pick a different issue.

**Comment by animator** [comment]
@badnikhil It is fine that you implement it now.

**Comment by badnikhil** [comment]
The response pane isn’t functional yet.working on getting it to display incoming messages cleanly.
https://github.com/user-attachments/assets/26ae9b65-8b88-4506-b77e-4bc28d756ad8

**Comment by animator** [comment]
> The response pane isn’t functional yet.working on getting it to display incoming messages cleanly. https://github.com/user-attachments/assets/26ae9b65-8b88-4506-b77e-4bc28d756ad8

cool

**Comment by TitanX311** [comment]
Is this issue still active?

**Comment by animator** [comment]
yes @TitanX311 .. to start with you can review the PR raised by @badnikhil

**Comment by roshan-melvin** [comment]
Hi @animator sir and @ashitaprasad mam,

Following up on the feedback and issue [#1529](https://github.com/foss42/apidash/pull/1529) , here is a video demonstration showcasing the implementation of both MQTT v3.1.1 and v5.0 protocols in API Dash for this issue:

https://github.com/user-attachments/assets/ae405bdb-545e-4c00-83f3-0db7e731b1c8

I’ve included this implementation in my new PR ([#1613](https://github.com/foss42/apidash/pull/1613)), which incorporates all the requested demo videos. Let me know if there are any specific MQTT features (like QoS, retained messages, or TLS configs) you'd like to see demonstrated further!

Thanks!
